bitcoin/src/blockdata/script/mod.rs

```diff
@@ -194,12 +194,45 @@ pub fn read_scriptint(v: &[u8]) -> Result<i64, Error> {
         }
     }
 
-    let (mut ret, sh) = v.iter().fold((0, 0), |(acc, sh), n| (acc + ((*n as i64) << sh), sh + 8));
+    Ok(scriptint_parse(v))
+}
+
+/// Decodes an integer in script format without non-minimal error.
+///
+/// The overflow error for slices over 4 bytes long is still there.
+/// See [`read_scriptint`] for a description of some subtleties of
+/// this function.
+pub const fn read_scriptint_non_minimal(v: &[u8]) -> Result<i64, Error> {
+    let len = v.len();
+    if len > 4 {
+        return Err(Error::NumericOverflow);
+    }
+    if len == 0 {
+        return Ok(0);
+    }
+
+    Ok(scriptint_parse(v))
+}
```

Comment:
Its no biggy, but if you like it I'd probably write this like
```rust
    if v.len() == 0 {
        return Ok(0);
    }
    if v.len() > 4 {
        return Err(Error::NumericOverflow);
    }
```

Success case first, error case second. Leaving out the local `len` var because the compiler will optimize that away.